grpc-cpp/plan.sh

```diff
@@ -0,0 +1,227 @@
+pkg_name=grpc-cpp
+pkg_distname=grpc
+pkg_origin=core
+pkg_version="1.19.1"
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=("Apache-2.0")
+pkg_source="https://github.com/grpc/grpc.git"
+pkg_shasum=noshasum
+
+pkg_deps=(
+    core/glibc
+    core/gcc-libs
+    core/zlib
+    core/openssl
+    core/protobuf
+    core/c-ares
+)
+pkg_build_deps=(
+    core/make
+    core/go
+    core/git
+    core/gcc
+    core/cmake
+    core/pkg-config
+    core/virtualenv
+    core/busybox-static
+)
+pkg_lib_dirs=(lib)
+pkg_include_dirs=(include)
+pkg_bin_dirs=(bin)
+pkg_description="A high performance, open-source universal RPC framework"
+pkg_upstream_url="https://grpc.io"
+
+do_begin() {
+  export HAB_ENV_CMAKE_FIND_ROOT_PATH_SEPARATOR=";"
+  export HAB_ENV_CMAKE_FIND_ROOT_PATH_TYPE="aggregate"
+}
+
+do_setup_environment() {
+  # making a cmake builddir that is the same as where the grpc tests expect it
+  set_buildtime_env BUILDDIR "cmake/build"
+
+  # this allows cmake users to utilize `CMAKE_FIND_ROOT_PATH` to find various cmake configs
+  push_runtime_env CMAKE_FIND_ROOT_PATH "${pkg_prefix}/lib/cmake/grpc"
+}
+
+do_download() {
```

Comment:
This is a suggestion, but one thing we can do for `git clone` downloads, is to use `do_unpack` instead of `do_download`.  This allows you to use the provided `do_clean`  and not have to manage that yourself.  

Ex: https://github.com/habitat-sh/core-plans/blob/master/dex/plan.sh#L27-L42

Reply:
Oh nice! That is brilliant. Good way to reduce the management of what folder I am in.